### DISCUSSION TITLE
Add table (widget table with specific query) in Step of Wizard

### DISCUSSION THREAD

**Discussion by HomaEEE** [filamentphp/filament#7857] [Help]
Hello, Can I add table or widget in step of wizard?

I want see some record by day in wizard for calculate working shift

![Снимок экрана 2023-08-17 в 17 48 57](https://github.com/filamentphp/filament/assets/1429343/579573fc-38fa-437f-9b72-7099f3ebffab)

**Comment by pboivin** [comment]
You can't currently embed a Table in a Form. This could change in the future.

**Comment by onursahindur** [comment]
I saw your question when I am trying to find an answer for this. 
I did some job via ai, and actually you can do it by this way:

Add a ViewField in the step of wizard as follows:
```
ViewField::make('previous_campaigns')
                    ->label('')
                    ->view('filament.company.pages.partials.previous-campaigns-table')
                    ->columnSpanFull(),
```

Make sure you implement HasTable.php with use InteractsWithTable.

```
class CreateDigitalAds extends CreateRecord implements HasTable
{
    use InteractsWithTable;

    protected static string $resource = DigitalAdsResource::class;
    ...
}
```

Then add table function to the Page:
   ```
protected function getPreviousCampaignsQuery(): Builder
    {
        return SomeBeautifulModel::query()
            ->where('uuid', $auth->id)
            ->latest('created_at');
    }

    public function table(Table $table): Table
    {
        return $table
            ->query($this->getPreviousCampaignsQuery())
            ->columns([
                TextColumn::make('title')
                    ->label('Title')
                    ->searchable(),
            ])
            ->paginated([5, 10, 25])
            ->defaultPaginationPageOption(5);
    }
```

I am still trying to figure out adding two different tables in the same wizard, if I find one, I will update my answer.
